### ISSUE TITLE
error: ‘BPF_PROG_TYPE_XRP’ undeclared

### ISSUE THREAD

**Issue by GorilaMond** [xrp-project/XRP#3]
Run as the guidance, but when running build_and_install_bpfkv.sh, I got an error:

> Building BPF-KV...
gcc -Wall -D_GNU_SOURCE -Wunused   -c -o helpers.o helpers.c
helpers.c: In function ‘load_bpf_program’:
helpers.c:137:31: error: ‘BPF_PROG_TYPE_XRP’ undeclared (first use in this function); did you mean ‘BPF_PROG_TYPE_XDP’?
  137 |     ret = bpf_prog_load(path, BPF_PROG_TYPE_XRP, &obj, &progfd);
      |                               ^~~~~~~~~~~~~~~~~
      |                               BPF_PROG_TYPE_XDP
helpers.c:137:31: note: each undeclared identifier is reported only once for each function it appears in
make: *** [<builtin>: helpers.o] Error 1

How can I fix it?

**Comment by yuhong-zhong** [comment]
It seems you are not in the XRP kernel. Could you please run `uname -r` to confirm the kernel version you are using?

**Comment by GorilaMond** [comment]
> It seems you are not in the XRP kernel. Could you please run `uname -r` to confirm the kernel version you are using?

Oh, it has been solved. After my exploration about the header file, I realized that the path `/usr/x86-64-linux-gnu/include` is prior to `/usr/include` when compliers are searching header files so that my complier has used declaration of the symbol in `/usr/x86-64-linux-gnu/include`. I deleted linux/bpf.h in `/usr/x86-64-linux-gnu/` for avoiding it.
